nixos/lib/test-driver/test-driver.py

```diff
@@ -525,6 +527,35 @@ def screenshot(self, filename: str) -> None:
             if ret.returncode != 0:
                 raise Exception("Cannot convert screenshot")
 
+    def copy_from_vm(self, source: str, target_dir: str = "") -> None:
+        """Copy a file from the VM (specified by an in-VM source path) to $out
+        (with an optional subdirectory parameter). The file is copied via the
+        shared directory.
+        """
+        # Compute the source, target, and intermediate shared file names
+        out_dir = pathlib.Path(os.environ.get("out", os.getcwd()))
+        src_orig = pathlib.Path(source)
+        int_shared_dir = pathlib.Path("/tmp/xchg")
+        with tempfile.TemporaryDirectory(dir=self.shared_dir) as shared_td:
+            shared_temp = pathlib.Path(shared_td)
+            shared_temp_in = int_shared_dir / shared_temp.name
+            src_in = shared_temp_in / src_orig.name
+            src_out = shared_temp / src_orig.name
+            tgt = out_dir / target_dir
```

Comment:
this is the absolute path of target_dir. I'd suggest to inline that or at least move if immediately before the actual copy action.